scripts/test/isis_powder/ISISPowderGemOutputTest.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import (absolute_import, division, print_function)
+
+from mantid.api import FileFinder
+import mantid.simpleapi as mantid  # Have to import Mantid to setup paths
+import unittest
+import tempfile
+import shutil
+import warnings
+
+from isis_powder.gem_routines import gem_output
+
+
+class ISISPowderGemOutputTest(unittest.TestCase):
+
+    _folders_to_remove = set()
+    CHECK_AGAINST = "BANK 1 100  25 RALF  -12266  96  -12266 -0.022 ALT                              \n"
+    GROUPING_SCHEME = [1,2]
+    IPF_FILE = "GEM_PF1_PROFILE.IPF"
+
+    def _find_file_or_die(self, name):
+        full_path = FileFinder.getFullPath(name)
+        if not full_path:
+            self.fail("Could not find file \"{}\"".format(name))
+        return full_path
+
+    def tearDown(self):
+        for folder in self._folders_to_remove:
+            try:
+                shutil.rmtree(folder)
+            except OSError as exc:
+                warnings.warn("Could not remove folder at \"{}\"\n"
+                              "Error message:\n{}".format(folder, exc))
+
+    def test_valid_save(self):
+        path_to_ipf = self._find_file_or_die(self.IPF_FILE)
+        temp_file_path = tempfile.mkdtemp()
+        self._folders_to_remove.add(temp_file_path)
+        wsgroup =[]
+        for i in range(2):
+            wsgroup.append(mantid.CreateSampleWorkspace(OutputWorkspace=str(i),
+                                                        NumBanks=1,
+                                                        BankPixelWidth=1,
+                                                        XMin=-0.5,
+                                                        XMax=0.5,
+                                                        BinWidth=0.01,
+                                                        XUnit='DSpacing',
+                                                        StoreInADS=True))
+
+
+
+        gem_output_test_ws_group = mantid.GroupWorkspaces(wsgroup)
+
+        gem_output.save_gda(d_spacing_group=gem_output_test_ws_group,
+                            gsas_calib_filename=path_to_ipf,
+                            grouping_scheme=self.GROUPING_SCHEME,
+                            output_path=temp_file_path + '\\test.gda')
+
+        self._find_file_or_die(temp_file_path + '\\test.gda')
+
+        with open(temp_file_path + '\\test.gda') as read_file:
+            first_line = read_file.readline()
+            self.assertEquals(first_line, self.CHECK_AGAINST)
```

Comment:
Why are we only checking the first line? Should we add a reference result and check against that?

Reply:
There's already a test that ensures saveGDA outputs the whole files correctly, this is mostly just to ensure that saveGDA is being called correctly, and not to verify that the entire format is correct.

Reply:
👍 Great.